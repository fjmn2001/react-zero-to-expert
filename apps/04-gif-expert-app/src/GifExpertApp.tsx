import { useState } from "react"
import AddCategory from "./category/AddCategory"
import GifGrid from "./category/GifGrid"

interface Props {
  defaultCategories?: string[]
}

const GifExpertApp = ({ defaultCategories = ["One Punch"] }: Props) => {
  const [categories, setCategories] = useState(defaultCategories)

  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  )
}

export default GifExpertApp
