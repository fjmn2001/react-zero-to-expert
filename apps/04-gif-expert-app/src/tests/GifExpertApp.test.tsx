import { describe, expect, test } from "vitest"
import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe("Testing <GifExpertApp/>", () => {
  test("it should show correctly", () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
  })

  test("it should show a list of GifGrid", () => {
    const defaultCategories = ["One Punch", "Samurai X"]
    const wrapper = shallow(
      <GifExpertApp defaultCategories={defaultCategories} />
    )
    expect(wrapper.find("GifGrid").length).toBe(2)
  })
})
