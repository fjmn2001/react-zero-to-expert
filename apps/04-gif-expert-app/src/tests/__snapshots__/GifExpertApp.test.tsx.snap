// Vitest Snapshot v1

exports[`Testing <GifExpertApp/> > it should show correctly 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
